WIP, ENH: Adds subpeak detection

rmarkello opened the pull request:
Closes #23.

Adds ability to detect subpeaks in clusters by specifying parameter `min_distance` in `create_output()` and `get_statmap_info()`, and adds relevant tests for this functionality. This function is _off_ by default, but I think that's potentially something that might be good for discussion.

Also fixes bugs in cluster detection where single clusters that had both positive + negative values were counted as part of the same cluster. Clustering is now performed separately on positives / negatives so they are counted as distinct clusters.

rmarkello commented:
I'm hoping this will pass, but please don't merge this quite yet! I have to update the CLI and add `min_distance` as an option to the argument parser. Also going to potentially beef up some doc-strings that I left rather bare.

rmarkello commented:
Pinging @miykael or @danjgale to see if they have thoughts on whether subpeak detection should happen by default or not!

danjgale commented:
Good catch on that +/- peak bug

danjgale commented:
RE: subpeaks, I personally think `min_distance` should default to `None`, which would only yield one peak per cluster. Then `if min_distance is not None`, subpeaks should be extracted accordingly.

rmarkello commented:
Sounds good, @danjgale! I agree that I think it's probably best left off by default, as parsing / interpreting the output with it on can require a bit more work. If you've got time to take a look at what I changed, that'd be awesome! If not, no worries.

I've added the CLI parameter as mentioned previously, but still want to beef up the doc-strings for all the helper functions I added for the subpeaks detection. @miykael: don't merge yet!

rmarkello commented:
Okay! So, this ended up being...a much larger pull request than intended. It amounts to a pretty sizable overhaul of the codebase. On the plus side, this _dramatically_ speeds everything up! I also added some tests and increased the coverage by a lot, so that's good, too!

I recognize that interpreting the changes_is a bit unwieldy, but I tried to refactor the code so that there's less of it! I think it can actually be reduced a bit more, but I think for now this is a good start.

Let me know if you have any thoughts or questions on what else might be implemented here! Otherwise, I'll leave it to either @miykael or @danjgale to merge.
